src/mono/mono/mini/mini-llvm.c

```diff
@@ -6189,11 +6189,11 @@ process_bb (EmitContext *ctx, MonoBasicBlock *bb)
 					}
 					break;
 				case LLVMArgWasmVtypeAsScalar:
-					if (!addresses [ins->sreg1]) {
+					if (!addresses [ins->sreg1] && lhs) {
```

Comment:
Shouldn't this case, when the methods just throw, be handled here? https://github.com/dotnet/runtime/pull/101299/files#diff-cc4f4d3a653b6ac0bc4a6a001e4f0321e94ce51cc16ec85dc18675d1f8976365R6124-R6131

Reply:
You are right, but the condition seems te be false in case of `LLVMArgWasmVtypeAsScalar` due to `src_in_reg=FALSE` in:
```c
if (src_in_reg && (!lhs || ctx->is_dead [ins->sreg1])) {
```

Do you think that we should update `case LLVMArgNormal: src_in_reg = TRUE; break;` to include `LLVMArgWasmVtypeAsScalar`?

Reply:
I think we should handle it similar to `LLVMArgVtypeAsScalar`, see my commit. (github didn't want me to allow to commit it as suggestion, so I committed it directly)

Then the rest of the changes should not be needed.

Reply:
Thanks for help!